apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -363,41 +362,36 @@ def meta(self) -> "MetadataWrapper":
     def ndim(self) -> int:
         return len(self._handle.dimension_names)
 
-    def _get_and_cast_domain(
-        self, domain_slot_getter: Callable[[str, DTypeLike], Tuple[object, object]]
+    def _cast_domainish(
+        self, domainish: List[Any]
     ) -> Tuple[Tuple[object, object], ...]:
         result = []
-        for name in self._handle.dimension_names:
-            dtype = self._handle.schema.field(name).type
-            if pa.types.is_timestamp(dtype):
-                np_dtype = np.dtype(dtype.to_pandas_dtype())
-                dom = domain_slot_getter(name, np_dtype)
+        for i, slot in enumerate(domainish):
```

Comment:
I might have missed it, but I don't think you actually use `i` in this loop.

```suggestion
        for slot in domainish:
```

Reply:
```
    index_column_names <- index_column_name_choices[[i]]
```

Reply:
and on #3032, more there